### ISSUE TITLE
ML Operability: Health Summary + Strict Config Visibility + Drift Error Standardization

### ISSUE THREAD

**Issue by jonkmatsumo** [jonkmatsumo/label-lag#385]
## Goal
coherent, bounded, and stable operational payloads for debugging and alerting

## In scope
- ML health summary attached to diagnostics
- strict-mode effective config in diagnostics/health
- standardized drift error fields (additive)
- golden shape tests
- docs

## Out of scope
- new infra
- new endpoints
- API breaking changes

## Acceptance criteria
- health payload stable + tested
- strict config visible + tested
- drift errors standardized + tested across failure modes
- ........................................ssssssssssssssssssssssssssssssss [ 17%]
sssssssssssssss......................................................... [ 35%]
.......................................................................s [ 53%]
s....................................................................... [ 71%]
........................................................................ [ 89%]
.....s...................................                                [100%]
=============================== warnings summary ===============================
python/tests/model/test_calibration_guards.py::TestCalibrationGuards::test_calibration_skipped_insufficient_samples
python/tests/model/test_calibration_guards.py::TestCalibrationGuards::test_calibration_skipped_insufficient_positives
python/tests/model/test_calibration_guards.py::TestCalibrationGuards::test_calibration_skipped_insufficient_negatives
python/tests/model/test_calibration_guards.py::TestCalibrationGuards::test_calibration_success_when_thresholds_met
  /Users/jonathan/git/label-lag/.venv/lib/python3.12/site-packages/xgboost/training.py:199: UserWarning: [13:53:32] WARNING: /Users/runner/work/xgboost/xgboost/src/learner.cc:790: 
  Parameters: { "use_label_encoder" } are not used.
  
    bst.update(dtrain, iteration=i, fobj=obj)

python/tests/model/test_calibration_guards.py: 4 warnings
python/tests/model/test_schema_hash.py: 1 warning
python/tests/test_hyperparameters.py: 4 warnings
python/tests/test_obs_training_identity.py: 1 warning
python/tests/test_train_feature_columns.py: 3 warnings
python/tests/test_train_metrics.py: 6 warnings
python/tests/test_tuning.py: 1 warning
  /Users/jonathan/git/label-lag/.venv/lib/python3.12/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details.
    warnings.warn(

python/tests/test_hyperparameters.py::TestHyperparamsLoggedToMlflow::test_hyperparams_logged_to_mlflow
python/tests/test_hyperparameters.py::TestDefaultHyperparamsUnchanged::test_default_hyperparams_unchanged
  /Users/jonathan/git/label-lag/.venv/lib/python3.12/site-packages/xgboost/training.py:199: UserWarning: [13:53:34] WARNING: /Users/runner/work/xgboost/xgboost/src/learner.cc:790: 
  Parameters: { "use_label_encoder" } are not used.
  
    bst.update(dtrain, iteration=i, fobj=obj)

python/tests/test_hyperparameters.py::TestEarlyStopping::test_early_stopping_logs_best_iteration
  /Users/jonathan/git/label-lag/.venv/lib/python3.12/site-packages/xgboost/callback.py:386: UserWarning: [13:53:34] WARNING: /Users/runner/work/xgboost/xgboost/src/learner.cc:790: 
  Parameters: { "use_label_encoder" } are not used.
  
    self.starting_round = model.num_boosted_rounds()

python/tests/test_train_feature_columns.py: 3 warnings
python/tests/test_train_metrics.py: 9 warnings
  /Users/jonathan/git/label-lag/.venv/lib/python3.12/site-packages/xgboost/training.py:199: UserWarning: [13:53:35] WARNING: /Users/runner/work/xgboost/xgboost/src/learner.cc:790: 
  Parameters: { "use_label_encoder" } are not used.
  
    bst.update(dtrain, iteration=i, fobj=obj)

python/tests/test_train_metrics.py: 8 warnings
python/tests/test_tuning.py: 2 warnings
  /Users/jonathan/git/label-lag/.venv/lib/python3.12/site-packages/xgboost/training.py:199: UserWarning: [13:53:36] WARNING: /Users/runner/work/xgboost/xgboost/src/learner.cc:790: 
  Parameters: { "use_label_encoder" } are not used.
  
    bst.update(dtrain, iteration=i, fobj=obj)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================ tests coverage ================================
______________ coverage: platform darwin, python 3.12.12-final-0 _______________

Name                                          Stmts   Miss Branch BrPart  Cover   Missing
-----------------------------------------------------------------------------------------
python/src/features/registry.py                  21      2      2      1    87%   65, 76
python/src/features/store.py                     81     59     14      0    23%   21, 26, 33, 42-51, 64-76, 79-81, 85-113, 119-155, 160
python/src/forecast/drift_cache.py               48      5     10      3    86%   30-33, 73, 79
python/src/forecast/model_manager.py            783    139    264     51    79%   83-88, 132->135, 185, 187, 190-191, 261->exit, 264-265, 280, 282->273, 285->293, 289->293, 294-298, 318-319, 378, 416->418, 421, 429->431, 434, 447-462, 470-472, 480, 487, 495, 498-501, 506, 528->531, 532->555, 534->544, 539->544, 547-553, 693, 787-796, 807-815, 938-939, 963, 969, 990->1014, 991->990, 999-1000, 1008, 1012-1013, 1024->1023, 1031, 1034-1035, 1046->1045, 1059-1060, 1075->1090, 1076->1075, 1088-1089, 1103->1117, 1104->1103, 1111, 1115-1116, 1122-1127, 1135-1137, 1143, 1147, 1153->1160, 1156-1159, 1192, 1281->1280, 1291-1292, 1317-1351, 1365->1364, 1367-1369, 1390-1394, 1416-1418, 1434-1462, 1477, 1491, 1494, 1540-1543
python/src/forecast/service.py                   91     24     14      2    75%   16-22, 45-47, 51->54, 71-73, 80-81, 119-121, 160-180
python/src/forecast/services.py                 224     52     68      9    76%   42-46, 59-61, 172-173, 315->320, 355-359, 368-375, 419-458, 470-479, 506, 510-511, 515-516, 520-521, 525-526, 547-549
python/src/inference/__init__.py                  5      1      2      1    71%   9
python/src/inference/config.py                   48     27     14      0    34%   26-30, 48-54, 58-61, 65-76
python/src/inference/logging.py                   3      3      0      0     0%   3-7
python/src/inference/server.py                   34     34      6      0     0%   3-64
python/src/inference/service.py                  54     15     10      3    72%   34, 42, 64-65, 71-74, 83-84, 96-97, 103-114
python/src/logging_util.py                       13     13      2      0     0%   3-72
python/src/main.py                               47     47      8      0     0%   3-133
python/src/model/evaluate.py                    140      5     20      3    95%   152, 306->exit, 425-429
python/src/model/loader.py                       98      3     34      2    96%   133->139, 139->144, 235-238
python/src/model/split_strategies.py             75      7     12      4    87%   35, 50-51, 79-80, 118, 126
python/src/model/train.py                       483     47    136     24    88%   54-56, 100-101, 105->114, 109-111, 121, 123, 128, 146-147, 278-283, 305-307, 335, 362->368, 374, 456, 458, 469-470, 472->475, 533->633, 601->605, 800, 812->827, 823, 834->856, 846->856, 852, 875, 974-982, 987-994
python/src/model/tuning.py                      310     59    132     30    77%   76-79, 85, 88, 91-92, 95, 100-101, 116-120, 128-135, 158, 169, 191->194, 358-359, 362->368, 377-378, 384, 386->exit, 397->399, 400-418, 469-474, 484-485, 487->493, 490-491, 514, 536->602, 543->551, 555->602, 596, 599-600, 603, 605, 631, 649->661, 652-660, 672, 675->678, 680, 681->664, 689->730, 727-728
python/src/training/audit.py                     90      3     20      2    95%   38->40, 63, 89-90
python/src/training/config.py                    12      1      0      0    92%   19
python/src/training/crud_client.py              222     58     52     12    72%   52-58, 65->70, 117-118, 126-127, 147-158, 166-171, 176-177, 185-188, 227, 231-232, 242-243, 252-253, 262-263, 280-281, 294-298, 308-311, 324->326, 327, 400, 402, 408-409, 425, 434-439, 446-447, 454, 467, 469, 494-495, 500-501, 513->515, 528-529
python/src/training/detect_drift.py             270     47     86     11    81%   84-90, 152, 231-232, 240, 242, 245-246, 333->341, 348, 350, 391, 415-421, 426-454, 490->492, 609-624, 628
python/src/training/gateway_client.py            19     13      4      0    26%   12-29, 35
python/src/training/gateway_grpc_client.py       76     57     16      0    21%   13-17, 26-31, 44-69, 83-120, 123-155, 158, 174, 185-193
python/src/training/grpc_errors.py                6      4      2      0    25%   13-17
python/src/training/job_queue.py                 30      1      4      1    94%   31
python/src/training/job_store.py                134     15     52     16    83%   31-32, 43-44, 61->exit, 63->exit, 65->exit, 69->exit, 76->exit, 78->exit, 86->exit, 88->exit, 90->exit, 101, 113, 145, 150-151, 167, 184-185, 191, 204, 221
python/src/training/materialize_features.py      26      6      4      1    77%   45-52
python/src/training/optuna_resume.py             74     18     20      7    71%   23, 31, 55, 65, 89-99, 103-106, 122-123, 147->156, 152-154
python/src/training/postgres_job_store.py       137     48     28      7    59%   27, 32, 40-41, 49, 159-166, 175, 190-240, 243-251, 263-280, 283->289, 286-287, 326, 329-344, 347, 391, 394
python/src/training/schemas.py                  311      3      4      0    98%   368-370
python/src/training/server.py                    52     52      4      0     0%   3-102
python/src/training/service.py                  618    304    218     54    49%   75-76, 79-80, 102-103, 111-112, 122, 125-126, 130-133, 141-142, 144->143, 146, 176, 178, 203, 212, 219->exit, 224-233, 237-488, 492-514, 518-577, 591-605, 607->610, 611-635, 640->661, 648-649, 653, 685->704, 726-728, 732-756, 768-783, 785->788, 789-802, 807->825, 815, 819, 826, 832, 838, 853, 884-964, 972-974, 980, 1011-1012, 1041-1045, 1053, 1055, 1064, 1133-1139, 1152-1158, 1165, 1168, 1172, 1181, 1194->1197, 1223, 1227, 1271, 1288->1290, 1303, 1307, 1309, 1323, 1333, 1357, 1359, 1363-1364, 1369, 1376, 1384->exit, 1389-1392, 1401-1419
python/src/training/tuning_startup.py            79      9     18      2    87%   31-35, 44-49, 53-58
python/src/training/worker.py                   220     36     60     16    80%   27, 38, 73-75, 79-82, 117, 129->134, 137-138, 140, 173, 187, 190, 201, 247-260, 276, 290-296, 301-302, 307, 328, 341->346, 351-352
-----------------------------------------------------------------------------------------
TOTAL                                          5161   1217   1350    262    74%

13 files skipped due to complete coverage.
351 passed, 50 skipped, 49 warnings in 25.27s green
